demonstrations/src/main/java/boofcv/demonstrations/segmentation/DemoSegmentColor.java

```diff
@@ -0,0 +1,277 @@
+	public void toggleClick() {
+		clickStatus = !clickStatus;
+	}
+
+	// color processing for the clicked pixel
+	public void processColor(int rgb){
+		ColorHsv.rgbToHsv((rgb >> 16) & 0xFF, (rgb >> 8) & 0xFF, rgb & 0xFF, color);
+		System.out.println("H = " + color[0] + " S = " + color[1] + " V = " + color[2]);
+	}
+
+	// default constructor for DemonstrationBase
+	public DemoSegmentColor( List<?> exampleInputs ) {
+		super(exampleInputs, ImageType.pl(3, GrayU8.class));
+
+		imageZoomPanel.setPreferredSize(new Dimension( 800, 400));
+		imageZoomPanel.getImagePanel().addMouseListener( new MouseAdapter() {
+			@Override
+			public void mouseClicked( MouseEvent e ) {
+				mouseX = e.getX();
+				mouseY = e.getY();
+				toggleClick();
+
+				//sanitizeMouseInput(); //not functioning correctly with zoom input
+
+				reprocessInput();
+			}
+		});
+		//imageZoomPanel.requestFocus();
+
+		imageZoomPanel.addMouseWheelListener( new MouseAdapter() {
+			@Override
+			public void mouseWheelMoved(MouseWheelEvent e) {
+				controls.setZoom(BoofSwingUtil.mouseWheelImageZoom(DemoSegmentColor.this.controls.zoom, e));
+			}
+		});
+
+
+		// configure the layout, Controls on the left, imageZoom on the main center panel
+		add(BorderLayout.WEST, controls);
+		add(BorderLayout.CENTER, imageZoomPanel);
+	}
+
+	@Override
+	protected void configureVideo( int which, SimpleImageSequence sequence ) {
+		super.configureVideo(which, sequence);
+		sequence.setLoop(true);
+	}
+
+	@Override
+	protected void handleInputChange(int source, InputMethod method, int width, int height) {
+		super.handleInputChange(source, method, width, height);
+		imageWidth = width;
+		imageHeight = height;
+
+		output = ConvertBufferedImage.checkDeclare(width,height,output, output.getType());
+		// change the size of the GUI to match the input image as well as handle zooming scale
+		BoofSwingUtil.invokeNowOrLater(() -> {
+			double zoom = BoofSwingUtil.selectZoomToShowAll(imageZoomPanel, width, height);
+			controls.setImageSize(width, height);
+			controls.setZoom(zoom);
+			imageZoomPanel.setScale(zoom);
+			imageZoomPanel.updateSize(width, height);
+			imageZoomPanel.getVerticalScrollBar().setValue(0);
+			imageZoomPanel.getHorizontalScrollBar().setValue(0);
+		});
+	}
+
+	// retrieve and process the user selected coordinates
+	public void clickProcessing( final BufferedImage image, int mouseX, int mouseY){
+
+		// accurate conversion of coordinates to image coordinates
+		Point2D_F64 imageCoordinates = imageZoomPanel.pixelToPoint(mouseX, mouseY);
+		double imageX = imageCoordinates.x;
+		double imageY = imageCoordinates.y;
+
+		// grab the rgb value at the image coordinates
+		int rgb = image.getRGB((int) imageX, (int) imageY);
+
+		//processes and populate color data array
+		processColor(rgb);
+	}
+
+	@Override
+	public void processImage( int sourceID, long frameID, BufferedImage buffered, ImageBase _input ) {
+
+		if(clickStatus == true){
+			clickProcessing(buffered, mouseX, mouseY);
```

Comment:
Took me a minute to figure out why videos were so weird. After you process the click the first time do not change the selected color!

Reply:
added boolean flag to prevent reprocessing if set.